### DISCUSSION TITLE
Is it possible to get the transaction id of transactions?

### DISCUSSION THREAD

**Discussion by pyper-instacart** [sqlalchemy/sqlalchemy#7990] [Usage Questions] [answered]
I want to be able to log it for troubleshooting purposes to co-relate any long running transactions we see in Postgres with logs.

**Comment by zzzeek** [comment, marked as answer]
see pg_current_xact_id() at https://www.postgresql.org/docs/current/functions-info.html


```py
current_xid = connection_or_session.scalar(text("select pg_current_xact_id()"))
```
